src/node_buffer.cc

```diff
@@ -541,6 +659,121 @@ void WriteDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness,
+         T (*strtoT)(const char*, char**, int)>
+uint32_t WriteInt64Generic(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
+
+  T val;
+  if (args[1]->IsNumber()) {
+    val = args[1]->IntegerValue();
```

Comment:
You don't do any range checks on the `{u}int` value passed from JS. Which means I could pass a negative number to `writeUint64()`. You'd then have a problem because `IntegerValue()` always returns an `int64_t`. Which means you'd be basically doing `uint64_t val = <some negative number>` (e.g. passing `-5` would become `18446744073709551611`).

IOW, you might as well only allow `int64_t` in because that's the largest integer value you'll be able to receive from JS.

@bnoordhuis is it UB to cast a negative number to unsigned that way?